### DISCUSSION TITLE
Testing retry interceptor

### DISCUSSION THREAD

**Discussion by acommodari** [nodejs/undici#2975] [Q&A]
Hello I'm using the new `Client.compose()` with `interceptors.retry` and I'm wondering how do I properly mock this for testing?

My existing tests with `MockAgent` seem to all be failing now

**Comment by mcollina** [comment]
Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.
